lib/mail.rb

```diff
@@ -9,26 +9,8 @@ module Mail # :doc:
   require 'net/smtp'
   require 'mini_mime'
 
-  if RUBY_VERSION <= '1.8.6'
-    begin
-      require 'tlsmail'
-    rescue LoadError
-      raise "You need to install tlsmail if you are using ruby <= 1.8.6"
-    end
-  end
-
-  if RUBY_VERSION >= "1.9.0"
-    require 'mail/version_specific/ruby_1_9'
-    RubyVer = Ruby19
-  else
-    require 'mail/version_specific/ruby_1_8'
-    RubyVer = Ruby18
-  end
-
```

Comment:
I'm afraid these are public constants that are broadly referenced, so removal will cause surprising downstream breakage: [RubyVer](https://github.com/search?q=%22Mail%3A%3ARubyVer%22&type=code), [Ruby18](https://github.com/search?q=%22Mail%3A%3ARuby18%22&type=code), [Ruby19](https://github.com/search?q=%22Mail%3A%3ARuby19%22&type=code)

A deprecation cycle would be sensible.

Reply:
Also `Ruby18`, given that we're dropping Ruby 1.8 support? The link you provided does not show any meaningful results in the first pages for me.